simulation/behavior_tree_plugin/src/action_node.cpp

```diff
@@ -208,8 +210,13 @@ boost::optional<std::string> ActionNode::getFrontEntityName(
   std::vector<std::string> entities;
   for (const auto & each : other_entity_status) {
     const auto distance = getDistanceToTargetEntityPolygon(spline, each.first);
-    if (distance) {
-      if (distance.get() < 40) {
+    const auto quat = quaternion_operation::getRotation(
+      entity_status.pose.orientation, other_entity_status.at(each.first).pose.orientation);
+    /**
+     * @note hard-coded parameter, if the Yaw value of RPY is in ~1.5708 -> 1.5708, entity is a candidate of front entity.
+     */
+    if (std::fabs(quaternion_operation::convertQuaternionToEulerAngle(quat).z) <= 1.5708) {
```

Comment:
Replace with `boost::math::constants::half_pi<T>()` instead of `1.5708` for accuracy.

Reply:
fixed in https://github.com/tier4/scenario_simulator_v2/pull/703/commits/06de0f8e1e453d3f4b79134044f853cd9d79a909